### PR TITLE
284 - Added OpenTelemetry Handler

### PR DESCRIPTION
**Pull request by momosh-ethernal**
Added OTel handler. Currently only dumping metric for number of peers from the DHT table.

### CODE REVIEW

File: src/network/event_loop.rs [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                    Err(err) => response_sender.send(Err(err.into())),
+                }
+            }
+            Command::Bootstrap { response_sender } => {
+                match self.swarm.behaviour_mut().kademlia.bootstrap() {
+                    Ok(query_id) => {
+                        self.pending_kad_queries
+                            .insert(query_id, QueryChannel::Bootstrap(response_sender));
+                    }
+                    // no available peers for bootstrap
```
Comment (sh3ll3x3c) [inline src/network/event_loop.rs:235]: Do we still need to deploy 2 clients initially?

**Review by sh3ll3x3c — APPROVED** [review]
lgtm

File: src/network/client.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            .await
+            .context("Sender not to be dropped")?
+    }
+
+    pub async fn bootstrap(&self) -> Result<()> {
+        // bootstrapping is impossible on an empty DHT table
+        // at least one node is required to be known, so check
+        let counted_peers = self.count_dht_entries().await?;
+        // for a bootstrap to succeed, we need at least 1 peer in our DHT
+        if counted_peers < 1 {
```
[the remark this answers is not in the document]
Comment (momosh-ethernal) [reply src/network/client.rs:49]: It doesn't count itself on cold starts

File: src/network/event_loop.rs (hunk 1 above)
Comment (momosh-ethernal) [reply src/network/event_loop.rs:267]: Not necessarily. Upon deployment, Bootstraps will wait for someone to dial them, and they will commence bootstraping after a successful connection.

File: src/main.rs [hunk 3]
```diff
@@ -150,50 +72,66 @@ async fn run() -> Result<()> {
...
+            if let Ok(multiaddr) = m_network_client.get_multiaddress().await {
+                if let Some(addr) = multiaddr {
+                    // set Multiaddress
+                    _ = ot_metrics.set_multiaddress(addr.to_string());
+                    if let Some(ip) = network::extract_ip(addr) {
+                        // set IP
+                        _ = ot_metrics.set_ip(ip);
+                    }
+                }
+            }
```
Comment (aterentic-ethernal) [inline src/main.rs:105]: It seems that there are two missing `await` calls:

```rust
            // try and read current multiaddress
            if let Ok(Some(addr)) = m_network_client.get_multiaddress().await {
                // set Multiaddress
                ot_metrics.set_multiaddress(addr.to_string()).await;
                if let Some(ip) = network::extract_ip(addr) {
                    // set IP
                    ot_metrics.set_ip(ip).await;
                }
            }

```

**Review by aterentic-ethernal — APPROVED** [review]

**Pull request merged**